app/views/Home/ProjectItem/index.tsx

```diff
@@ -161,6 +283,24 @@ function ProjectItem(props: RecentProjectItemProps) {
             )}
             headerActions={(
                 <>
+                    {isDefined(projectId) && (isPinned ? (
+                        <QuickActionButton
+                            name={projectId}
+                            onClick={handleUnpinProject}
+                            title="Unpin this project"
+                        >
+                            <RiUnpinFill />
+                        </QuickActionButton>
+                    ) : (
+                        <QuickActionButton
+                            name={projectId}
+                            onClick={handlePinProject}
+                            title={disablePinButton ? 'You can only pin 5 projects.' : 'Pin this project'}
+                            disabled={disablePinButton}
+                        >
+                            <RiPushpinFill />
+                        </QuickActionButton>
```

Comment:
I don't think this will work.
If projectId is not defined, unpin will be shown, but I think both should be hidden.